ui/src/containers/configs/EditConfig.js

```diff
@@ -356,612 +356,703 @@ class EditConfig extends Component {
+                          pipelineOptions.filter(
+                            (option) => option.value === config.spec["pipeline"]
+                          )[0]
+                        }
+                        options={pipelineOptions}
+                        onChange={(value) => {
+                          this.handleChange(
+                            "pipeline",
+                            value.value,
+                            "deployment.spec"
+                          );
+                        }}
+                      />
+                    </div>
+                    <div className="col-12 mt-2">
+                      <label className="form-label">Replicas</label>
+                      <input
+                        type="number"
+                        className="form-control"
+                        id="replicas"
+                        min="0"
+                        name="replicas"
+                        onChange={(value) => {
+                          this.handleChange(
+                            "replicas",
+                            value.target.value,
+                            "deployment.spec.replicas"
+                          );
+                        }}
+                        placeholder="1"
+                        step="1"
+                        value={
+                          !isNaN(this.state.config.spec["replicas"])
+                            ? this.state.config.spec["replicas"]
+                            : ""
+                        }
+                      />
+                    </div>
+                  </>
                 )}
-                <div className="col-12 mt-4">
-                  <a
-                    href={`/configs/${config.uuid}`}
-                    className="btn btn-primary text-white mb-4"
-                    onClick={this.handleUpdateConfig}
-                  >
-                    Update config
-                  </a>
-                </div>
+              </>
+            )}
+            {![undefined, ""].includes(this.state.errorMessage) && (
+              <div className="alert alert-danger mt-4">
+                <p className="h6 fs-bolder">API response:</p>
+                {this.state.errorMessage}
               </div>
+            )}
+            <div className="col-12 mt-4 mb-4">
+              <a
+                href={`/configs/${config.uuid}`}
+                className="btn btn-primary text-white"
+                onClick={this.handleUpdateConfig}
+              >
+                Update config
+              </a>
+              <button
+                className="btn btn-outline-primary ms-2"
+                onClick={this.toggleForm}
+              >
+                Edit as JSON
+              </button>
             </div>
-          </form>
+          </div>
+        </div>
+      </form>
+    );
+  }
+
+  toggleForm(event) {
+    event.preventDefault();
+    let isShowingPayloadEditor = !this.state.showPayloadEditor;
```

Comment:
I am a bit confused here. Why do we need to negate `this.state.showPayloadEditor`?

Reply:
The if conditions below are considering, whether we move to or away from the editor. That way it was possible to avoid some branching and use `isShowingPayloadEditor` also as the next desired state. If you have suggestions on how to improve readability in context of the if statement below, happy to refactor.